dataflow/run_template/README.md

```diff
@@ -0,0 +1,137 @@
+# Run template
+
+[`main.py`](main.py) - Script to run an [Apache Beam] template on [Google Cloud Dataflow].
+
+The following examples show how to run the [`Word_Count` template], but you can run any other template.
+
+For the `Word_Count` template, we require to pass an `output` Cloud Storage path prefix, and optionally we can pass an `inputFile` Cloud Storage file pattern for the inputs.
+If `inputFile` is not passed, it will take `gs://apache-beam-samples/shakespeare/kinglear.txt` as default.
+
+## Before you begin
+
+1. Install the [Cloud SDK].
+
+1. [Create a new project].
+
+1. [Enable billing].
+
+1. [Enable the APIs](https://console.cloud.google.com/flows/enableapi?apiid=dataflow,compute_component,logging,storage_component,storage_api,bigquery,pubsub,datastore.googleapis.com,cloudfunctions.googleapis.com,cloudresourcemanager.googleapis.com): Dataflow, Compute Engine, Stackdriver Logging, Cloud Storage, Cloud Storage JSON, BigQuery, Pub/Sub, Datastore, Cloud Functions, and Cloud Resource Manager.
+
+1. Setup the Cloud SDK to your GCP project.
+
+   ```bash
+   gcloud init
+   ```
+
+1. Create a Cloud Storage bucket.
+
+   ```bash
+   gsutil mb gs://your-gcs-bucket
+   ```
+
+## Setup
+
+The following instructions will help you prepare your development environment.
+
+1. [Install Python and virtualenv].
+
+1. Clone the `python-docs-samples` repository.
+
+    ```bash
+    git clone https://github.com/GoogleCloudPlatform/python-docs-samples.git
+    ```
+
+1. Navigate to the sample code directory.
+
+   ```bash
+   cd python-docs-samples/dataflow/run_template
+   ```
+
+1. Create a virtual environment and activate it.
+
+   ```bash
+   python3 -m virtualenv env
```

Comment:
Our other samples use "virtualenv env" here -- perhaps we should use that by default, but if this is Python 3 only and you feel that that will cause difficulties for the average user, we could add a comment suggesting "virtualenv env -p python3" for users on Python 2-by-default platforms (which should become uncommon after the end of this year)

Refer to e.g. https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/bigtable/quickstart for an example. Also, note that the example I linked uses our README preprocessor; it's not a requirement but you can generate READMEs using README.rst.in if that is helpful for you and will cut down on boilerplate.

Reply:
I'm changing to `virtualenv env` for now. If at a certain point in the future it makes sense to use the generator I'll switch to it, but I think I'll keep it as is for the time being. Thanks!